doc/classes/@GlobalScope.xml

```diff
@@ -2773,7 +2773,24 @@
 		</constant>
 		<constant name="PROPERTY_HINT_TYPE_STRING" value="23" enum="PropertyHint">
 			If a property is [String], hints that the property represents a particular type (class). This allows to select a type from the create dialog. The property will store the selected type as a string.
+		</constant>
+		<constant name="PROPERTY_HINT_NODE_PATH_TO_EDITED_NODE" value="24" enum="PropertyHint">
+		</constant>
+		<constant name="PROPERTY_HINT_OBJECT_TOO_BIG" value="25" enum="PropertyHint">
+		</constant>
+		<constant name="PROPERTY_HINT_NODE_PATH_VALID_TYPES" value="26" enum="PropertyHint">
+		</constant>
+		<constant name="PROPERTY_HINT_SAVE_FILE" value="27" enum="PropertyHint">
+		</constant>
+		<constant name="PROPERTY_HINT_GLOBAL_SAVE_FILE" value="28" enum="PropertyHint">
+		</constant>
+		<constant name="PROPERTY_HINT_INT_IS_OBJECTID" value="29" enum="PropertyHint">
+		</constant>
+		<constant name="PROPERTY_HINT_INT_IS_POINTER" value="30" enum="PropertyHint">
+		</constant>
+		<constant name="PROPERTY_HINT_ARRAY_TYPE" value="31" enum="PropertyHint">
 			If a property is [Array], hints the editor how to show elements. The [code]hint_string[/code] must encode nested types using [code]":"[/code] and [code]"/"[/code].
```

Comment:
So the description was in its correct place, and this is wrong. However, the usage of `PROPERTY_HINT_TYPE_STRING` is pretty unclear. It's only used with arrays in the `PackedScene` parser. In the inspector we only support it for strings.

See https://github.com/godotengine/godot/pull/77734. cc @dalexeev

Reply:
> So the description was in its correct place, and this is wrong.

It not.

`PROPERTY_HINT_TYPE_STRING` is not used for `Array` properties, only for `String`s. And it seems to be used only for a single property in the Scene importer.

`Array` hint which is using described format is `PROPERTY_HINT_ARRAY_TYPE`.

Reply:
Seems like `@export_multiline` can set it for array and with the same, but inspector won't parse it for array, so it should be broken. I guess there are two hits for the same thing for some reason.

Reply:
https://github.com/godotengine/godot/pull/38063 implemented `PROPERTY_HINT_ARRAY_TYPE`. We don't support it in the inspector though.

Reply:
It kinda supported, but inspector only check for hint string, not actual hint:
https://github.com/godotengine/godot/blob/c131bf9973a1a8b585f3d0dc59945c2b6fffa30a/editor/editor_properties_array_dict.cpp#L595-L614

Reply:
`PROPERTY_HINT_ARRAY_TYPE` expects an element type as a string (e.g. `int`, `Node`) and is used in GDScript as well. `PROPERTY_HINT_TYPE_STRING` allows encoding element type with `hint` and `hint_string`, supports nested arrays, and is not checked in GDScript.

The inspector doesn't check `hint` for arrays at all, only `hint_string`, and this is probably a bug since it allows you to use the wrong `hint`.

See also #77618.

Reply:
> PROPERTY_HINT_ARRAY_TYPE expects an element type as a string (e.g. int, Node) and is used in GDScript as well.

It most places, it is set via `MAKE_RESOURCE_TYPE_HINT` macro, which is using `int` for a `subType` in `subType/subTypeHint:subTypeHintString`. Inspector also parses sub-type part as `int`. But there are plenty of places using a single string sub-type, without the hint part. So it's even bigger mess.

Reply:
This is a bug. See:

https://github.com/godotengine/godot/blob/332bc469c46e1372cdf3222bd3d31c382afc2784/core/variant/typed_array.h#L154-L188

https://github.com/godotengine/godot/blob/332bc469c46e1372cdf3222bd3d31c382afc2784/core/doc_data.cpp#L58-L59

https://github.com/godotengine/godot/blob/332bc469c46e1372cdf3222bd3d31c382afc2784/editor/doc_tools.cpp#L523-L524

https://github.com/godotengine/godot/blob/332bc469c46e1372cdf3222bd3d31c382afc2784/modules/gdscript/gdscript_analyzer.cpp#L4723-L4725